src/main/java/core/WebDriverManagerSingleton.java

```diff
@@ -1,48 +1,104 @@
-    public static WebDriver driver;
+    public WebDriver driver;
+    public WebDriverWait webDriverWait;
+    private static WebDriverManagerSingleton instance = null;
+
+    /**
+     * Constructor of page WebDriverManagerSingleton.
+     */
+    public WebDriverManagerSingleton() {
+        initialize();
+    }
 
     /**
      * Gets an instance of WebDriver according to browser.
      *
      * @param browser enum with the browser name.
      * @return an instance of WebDriver
      */
-    public static WebDriver getWebDriver(Browser browser) {
-        if (driver == null) {
-            if (browser == Browser.CHROME) {
-                WebDriverManager.chromedriver().setup();
+    public WebDriver getWebDriver(String browser) {
+        switch (browser) {
+            case "CHROME":
                 driver = new ChromeDriver();
-            }
-            if (browser == Browser.EDGE) {
-                WebDriverManager.edgedriver().setup();
-                driver = new EdgeDriver();
-            }
-            if (browser == Browser.FIREFOX) {
-                WebDriverManager.firefoxdriver().setup();
+                break;
+            case "FIREFOX":
                 driver = new FirefoxDriver();
-            }
+                break;
+            case "EDGE":
+                driver = new EdgeDriver();
+                break;
+            default:
+                driver = null;
+        }
+
+        return driver;
+    }
+
+    /**
+     * Initializes the settings for the driver.
+     */
+    public  void initialize() {
+        DriverManagerType driverManagerType = DriverManagerType.valueOf(SetUp.BROWSER.getValue());
+        WebDriverManager.getInstance(driverManagerType).setup();
+        this.driver = getWebDriver(SetUp.BROWSER.getValue());
+        this.driver.manage().window().maximize();
+        this.driver.manage().timeouts()
+                .implicitlyWait(Long.parseLong(SetUp.WAIT_TIME_VALUE.getValue()), TimeUnit.SECONDS);
+        webDriverWait = new WebDriverWait(driver, Long.parseLong(SetUp.WAIT_TIME_VALUE.getValue()),
+                Long.parseLong(SetUp.WAIT_SLEEP_TIME.getValue()));
+    }
+
+    /**
+     * Gets Instance of WebDriverManagerSingleton.
+     *
+     * @return instance of WebDriverManagerSingleton.
+     */
+    public static WebDriverManagerSingleton getInstance() {
+        if (instance == null || instance.driver == null) {
+            instance = new WebDriverManagerSingleton();
         }
-        driver.manage().window().maximize();
+        return instance;
+    }
+
+    /**
+     * Gets the WebDriver.
+     *
+     * @return the WebDriver.
+     */
+    public WebDriver getDriver() {
         return driver;
     }
 
-    public static void closeDriver() {
+    /**
+     * Gets the WebDriverWait.
+     *
+     * @return the WebDriverWait.
+     */
+    public WebDriverWait getWebDriverWait() {
+        return webDriverWait;
+    }
+
+    public void closeDriver() {
```

Comment:
Java doc is missing is some methods of this class

Reply:
Thanks, I have added that